lib/debugger/clients/explorerScripts.ts

```diff
@@ -0,0 +1,88 @@
+import {promise, WebElement} from 'selenium-webdriver';
+
+import {ProtractorBrowser} from '../../browser';
+import {Locator} from '../../locators';
+let absoluteXPath = require('./explorerXpath').absoluteXPath;
+
+export class ExplorerScripts {
+  static list(locator: Locator) {
+    (<any>global).tempRootEl = (<any>global).browser.rootEl;
+    return (<any>global)
+        .browser.findElements(locator)
+        .then((arr: WebElement[]) => {
+          let found: string[] = [];
+          for (let i = 0; i < arr.length; ++i) {
+            arr[i].getText().then((text: string) => { found.push(text); });
+          }
+          return found;
+        })
+        .catch((err: Error) => { throw err; });
+  }
+
+  static highlight(locator: Locator) {
+    let xPaths: string[] = [];
+    let cssBorders: string[] = [];
+
+    return (<any>global)
+        .browser.findElements(locator)
+        .then((arr: WebElement[]) => {
+
+          for (let i = 0; i < arr.length; ++i) {
+            ExplorerScripts.getAbsoluteXPath(arr[i]).then(
```

Comment:
I think we can get around the whole xpath business by passing in the web element. Webdriver's `ExecuteScript` command lets you pass in a WebElement as an argument, which is converted to the proper HTMLElement on the browser side.